source/fe/fe_bernardi_raugel.cc

```diff
@@ -177,6 +177,80 @@ FE_BernardiRaugel<dim>::initialize_support_points()
     }
 }
 
+
+
+template <int dim>
+void
+FE_BernardiRaugel<dim>::fill_fe_values(
+  const typename Triangulation<dim, dim>::cell_iterator &cell,
+  const CellSimilarity::Similarity                       cell_similarity,
+  const Quadrature<dim> &                                quadrature,
+  const Mapping<dim, dim> &                              mapping,
+  const typename Mapping<dim, dim>::InternalDataBase &   mapping_internal,
+  const dealii::internal::FEValuesImplementation::MappingRelatedData<dim, dim>
+    &                                                       mapping_data,
+  const typename FiniteElement<dim, dim>::InternalDataBase &fe_internal,
+  dealii::internal::FEValuesImplementation::FiniteElementRelatedData<dim, dim>
+    &output_data) const
+{
+  std::cout << "fill_fe_values called!" << std::endl;
+  std::cout << "Calling FE_PolyTensor<dim, dim>::fill_fe_values" << std::endl;
+
+  // Call the base implementation then we'll touch up the data slightly
+  FE_PolyTensor<dim, dim>::fill_fe_values(
+     cell,
+     cell_similarity,
+     quadrature,
+     mapping,
+     mapping_internal,
+     mapping_data,
+     fe_internal,
+     output_data);
+  std::cout << "Length of mapping_data.normal_vectors: " << mapping_data.normal_vectors.size() << std::endl;
+
+  // Convert to the correct internal data class for this FE class.
+  // Assert(dynamic_cast<const InternalData *>(&fe_internal) != nullptr,
+  //    ExcInternalError());
```

Comment:
you can put that assertion in again.

Reply:
I tried putting it back in, but it threw an error
```
/home/graham/Programming/cpp/Deal.II/dealii-source/source/fe/fe_bernardi_raugel.cc:209:29: error: ISO C++ forbids declaration of ‘type name’ with no type [-fpermissive]
  209 |   Assert(dynamic_cast<const InternalData *>(&fe_internal) != nullptr,
```
I think it already checks this when I call `FE_PolyTensor<dim, dim>::fill_fe_values`, so I think it should be safe to remove.